launch_ros/launch_ros/actions/load_composable_nodes.py

```diff
@@ -230,19 +233,48 @@ def execute(
         # Generate load requests before execute() exits to avoid race with context changing
         # due to scope change (e.g. if loading nodes from within a GroupAction).
         load_node_requests = []
+        autostart_actions = []
         for node_description in self.__composable_node_descriptions:
             request = get_composable_node_load_request(node_description, context)
             # The request can be None if the node description's condition evaluates to False
             if request is not None:
                 load_node_requests.append(request)
 
+            # If autostart is enabled, transition to the 'active' state.
+            if node_description.node_autostart:
+                complete_node_name = request.node_namespace + request.node_name
+                if not complete_node_name.startswith('/'):
+                    complete_node_name = '/' + complete_node_name
+                self.__logger.info(
+                    'Autostart enabled for requested lifecycle node {}'.format(complete_node_name))
+                node_description.init_lifecycle_event_manager(complete_node_name, context)
+                autostart_actions.append(
+                    LifecycleTransition(
+                        lifecycle_node_names=[complete_node_name],
+                        transition_ids=[lifecycle_msgs.msg.Transition.TRANSITION_CONFIGURE]
+                    ))
+                autostart_actions.append(
+                    LifecycleTransition(
+                        lifecycle_node_names=[complete_node_name],
+                        transition_ids=[lifecycle_msgs.msg.Transition.TRANSITION_ACTIVATE]
+                    ),
+                )
+
         if load_node_requests:
             context.add_completion_future(
                 context.asyncio_loop.run_in_executor(
                     None, self._load_in_sequence, load_node_requests, context
                 )
             )
 
+        load_actions = super().execute(context)
+        if load_actions is not None and len(autostart_actions) != 0:
+            return load_actions + autostart_actions
+        if load_actions is not None:
+            return load_actions
+        if len(autostart_actions) != 0:
+            return autostart_actions
```

Comment:
I think this could just be:

```suggestion
        return autostart_actions
```

But if you want to call out to super I think that's ok. We just know (at least for now) that `launch.Action` always returns `None`:

https://github.com/ros2/launch/blob/a93ce70943c65b938d38eab78322dc27722b7a6f/launch/launch/action.py#L123-L129

But that being said, in case someone changes the base class of this class and doesn't notice this part of the code, calling super defensively isn't a bad idea.

Leave it or change it, doesn't matter to me, just wanted to point it out.

Reply:
I am copying the styling of the existing codebase here, this is how its done in all other locations I viewed, so I'd say we should leave it.